File: 238-product-of-array-except-self/product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        left = [1]
        right = [1]
        for i in range(len(nums)):
            left.append(left[-1] * nums[i])
            right.append(right[-1] * nums[-i - 1])
        right = right[::-1]
        ans = []
        for i in range(1, len(nums) + 1):
            ans.append(left[i-1] * right[i])
        
        return ans

File: 238-product-of-array-except-self/product-of-array-except-self.go
func productExceptSelf(nums []int) []int {
    sufix := []int{1,nums[len(nums)-1]}
    n := len(nums)-2
    for n >= 0 {
        temp := sufix[len(sufix)-1] * nums[n]
        sufix = append(sufix,temp)
        n -= 1
    }
    ans := []int{}
    _mul := 1
    n = len(sufix)-2
    for i,x := range nums{
        temp := sufix[n-i] * _mul
        _mul *= x
        ans = append(ans,temp)
        
    }
    return ans

}